setup.py

```diff
@@ -23,6 +23,7 @@
         'typing_inspect>=0.3.1',
     ],
     extras_require={
+        ':python_version<"3.2"': ['backports.functools_lru_cache'],
```

Comment:
Not sure if this change is really worth adding another install dependency for 2.7. I'm not planning to support any EOL Python versions past 2020-01-01, which means we can drop anything below 3.5. What do you think?

Reply:
I have stopped using py2 years ago, so I basically don't care about it.  We can just delay this PR until 2020-01-01, or drop py2 support now, or put in this PR as is for now, whatever you prefer.

Reply:
I'd rather hold off on this for the moment. Once I've tagged and published 5.2, I'd be happy to start making backwards-incompatible changes on master in preparation for 6.0. If necessary, we can branch off 5.2 to make bug fixes on request.

Reply:
OK, let's just leave this as is for now.